packages/core/src/provider.tsx

```diff
@@ -54,7 +54,7 @@ export function Web3ReactProvider({
   // because we're calling `getPriorityConnector` with these connectors, we need to ensure that they're not changing in place
   if (
     connectors.length != cachedConnectors.current.length ||
-    connectors.some((connector, i) => connector !== cachedConnectors.current[i])
+    connectors.some(([connector], i) => connector !== cachedConnectors.current[i][0])
```

Comment:
can you name `cachedConnectors.current[i][0]` as `currentConnector` and have `[connector]` be `[nextConnector]`?

Reply:
really it should be `connector` and `previousConnector/cachedConnector` imo